### PR TITLE
Bug/update type

### PR DESCRIPTION
**Pull request by MRyderOC**
There are 4 types that require the mask parameter to be passed in order to update using the whole protobuf obj:
- Environment
- Experiment
- TestCase
- Version

the `_update_kwargs` method in `ScrapiBase` is designed to take care of these special cases.

### CODE REVIEW

**Comment by MRyderOC** [comment]
The update methods are only updated for TestCase and Environment classes since we don't have update methods for Version and Experiments currently.

File: src/dfcx_scrapi/core/environments.py [hunk 1]
```diff
@@ -293,44 +293,35 @@ def create_environment_by_display_name(
     def update_environment(
         self,
         environment_id: str,
-        environment_obj: types.Environment = None,
+        obj: types.Environment = None,
```
Comment (kmaphoenix) [inline src/dfcx_scrapi/core/environments.py:296]: @MRyderOC can we revert this change and leave `environment_obj` the same?  

Changing arg names will break backwards compat for anyone already running this code.  
Even if the arg name is non-standard, we don't want to break any existing user workflows.

File: src/dfcx_scrapi/core/scrapi_base.py [hunk 2]
```diff
@@ -396,6 +398,51 @@ def get_api_calls_count(self) -> int:
...
+                "test_case_conversation_turns", "creation_time",
+                "last_test_result",
+            ],
+            "Version": [
+                "name", "display_name", "description", "nlu_settings",
+                "create_time", "state",
+            ],
+        }
+        if isinstance(obj, types.Environment):
+            paths = attrs_map["Environment"]
```
Comment (kmaphoenix) [inline src/dfcx_scrapi/core/scrapi_base.py:430]: @MRyderOC how do these work if I only want to update 1 single kwarg? For example if I only pass in `display_name` but you set `["name", "display_name", "description", "nlu_settings", "create_time", "state",]`.

Does this null out the other data? or only update the 1 kwarg?

**Review by kmaphoenix — CHANGES_REQUESTED** [review]

File: src/dfcx_scrapi/core/scrapi_base.py (hunk 2 above)
Comment (MRyderOC) [reply src/dfcx_scrapi/core/scrapi_base.py:430]: Well in this method we check for kwargs first and if there is any we use that for updating: https://github.com/GoogleCloudPlatform/dfcx-scrapi/blob/4ec03f79b58653352874181992f7c6a8cf1c10cd/src/dfcx_scrapi/core/scrapi_base.py#L404-L407

Comment (kmaphoenix) [reply src/dfcx_scrapi/core/scrapi_base.py:430]: Discussed offline and understand the process here now! great solution! thanks!

**Review by kmaphoenix — APPROVED** [review]
LGTM
